websites/M/ManiaExchange/presence.ts

```diff
@@ -0,0 +1,451 @@
+
+    if (currentPath[0] === "auth") {
+      if (currentPath[1] === "login") {
+        presenceData.details = "Logging in";
+      } else if (
+        currentPath[1] === "register" ||
+        currentPath[1] === "resend_confirm"
+      ) {
+        presenceData.details = "Registering an account";
+      } else if (currentPath[1] === "forgot") {
+        presenceData.details = "Figuring out the password";
+      }
+    } else if (currentPath[0] === "user") {
+      presenceData.details = "Configuring their account";
+    }
+  } else if (currentURL.hostname.startsWith("tmtube")) {
+    updateCallback.function = (): void => {
+      presenceData.smallImageKey = "tmtube";
+      presenceData.smallImageText = "TMTube Archive";
+      if (currentPath[0] === "") {
+        presenceData.details = "On the home page";
+      } else if (currentPath[0] === "view") {
+        presenceData.details = document.querySelector("h2").textContent.trim();
+        presenceData.state = document
+          .querySelector(".box-user h2")
+          .textContent.trim();
+        try {
+          if (
+            document
+              .querySelector(".mejs__playpause-button button")
+              .getAttribute("aria-label") === "Pause"
+          ) {
+            const video: HTMLVideoElement = document.querySelector("video");
+            const timestamps = getTimestamps(
+              Math.floor(video.currentTime),
+              Math.floor(video.duration)
+            );
+            presenceData.startTimestamp = timestamps[0];
+            presenceData.endTimestamp = timestamps[1];
+          } else {
+            delete presenceData.startTimestamp;
+          }
+        } catch (e) {
+          delete presenceData.startTimestamp;
+        }
+      } else if (currentPath[0] === "search") {
+        presenceData.details = "Searching for a video";
+        presenceData.state = getURLParam("query");
+      }
+    };
+  } else if (currentURL.hostname.startsWith("api")) {
+    presenceData.smallImageKey = "api";
+    presenceData.smallImageText = "API Documentation";
+
+    if (currentPath[0] === "") {
+      presenceData.details = "On the home page";
+    } else if (currentPath[0] === "documents") {
+      presenceData.details = "Viewing a page";
+      switch (currentPath[1]) {
+        case "reference":
+          presenceData.state = "API Reference";
+          break;
+        case "enum":
+          presenceData.state = "Enumeration values";
+          break;
+        case "conventions":
+          presenceData.state = "Conventions";
+          break;
+        case "changes":
+          presenceData.state = "Changes";
+          break;
+      }
+    }
+  }
+})();
+
+if (updateCallback.present) {
+  presence.on("UpdateData", async () => {
+    resetData();
+    updateCallback.function();
+    console.log(JSON.stringify(presenceData));
+    presence.setActivity(presenceData);
+  });
+} else {
+  console.log(presenceData);
```

Comment:
remove console log

Reply:
Will do. There aren't supposed to be an else block on that but I forgot about it.